src/Symfony/Component/Config/Util/XmlUtils.php

```diff
@@ -37,9 +37,14 @@ private function __construct()
      * @return \DOMDocument
      *
      * @throws \InvalidArgumentException When loading of XML file returns error
+     * @throws \RuntimeException         When DOM extension is missing
      */
     public static function loadFile($file, $schemaOrCallable = null)
     {
+        if (!extension_loaded('dom')) {
```

Comment:
you pointed out that `dom` doesn't guarantee `xml`. Should we put `|| !extension_loaded('xml')` since this is calling xml functions?

Reply:
yes, `dom` does not guarantee `xml`, but php itself guarantees libxml, so all `libxml_*` will work, unless php is compiled from source with `--disable-libxml` (which is probably where uncommon)

```
$ apt-cache depends php7.0-fpm
php7.0-fpm
  Depends: libmagic1
  Depends: mime-support
  Depends: php7.0-cli
  Depends: php7.0-common
  Depends: php7.0-json
  Depends: php7.0-opcache
  Depends: tzdata
  Depends: ucf
  Depends: init-system-helpers
  Depends: lsb-base
  Depends: libapparmor1
  Depends: libc6
  Depends: libpcre3
  Depends: libssl1.0.0
  Depends: libsystemd0
  Depends: libxml2
  Depends: zlib1g
  Conflicts: <php5-fpm>
  Suggests: php-pear
```
```
$ apt-cache depends php7.0-cli
php7.0-cli
  Depends: libedit2
  Depends: libmagic1
  Depends: mime-support
  Depends: php7.0-common
  Depends: php7.0-json
  Depends: php7.0-opcache
  Depends: php7.0-readline
  Depends: tzdata
  Depends: ucf
  Depends: libc6
  Depends: libpcre3
  Depends: libssl1.0.0
  Depends: libxml2
  Depends: zlib1g
  Breaks: <php5-cli>
  Suggests: php-pear
  Replaces: <php5-cli>
```
```
$ apt-cache depends php7.0-cgi
php7.0-cgi
  Depends: libmagic1
  Depends: mime-support
  Depends: php7.0-cli
  Depends: php7.0-common
  Depends: php7.0-json
  Depends: php7.0-opcache
  Depends: tzdata
  Depends: ucf
  Depends: libc6
  Depends: libpcre3
  Depends: libssl1.0.0
  Depends: libxml2
  Depends: zlib1g
  Conflicts: <php5-cgi>
  Suggests: php-pear
```

Reply:
ok then :)